recipes/libpq/all/conanfile.py

```diff
@@ -64,6 +64,8 @@ def _configure_autotools(self):
             args = ['--without-readline']
             args.append('--with-zlib' if self.options.with_zlib else '--without-zlib')
             args.append('--with-openssl' if self.options.with_openssl else '--without-openssl')
+            if tools.cross_building(self.settings) and not self.options.with_openssl:
+                args.append("--disable-strong-random")
```

Comment:
I this flag the appropriate fix for the issue or is this just a workaround? If it is a workaround I would like to have a comment here explaining the case or a link to an issue with the explanation

Reply:
There are two ways for libpq to get random numbers (see: https://github.com/postgres/postgres/blob/00b047fa67e9f4428a682bd90d5e6b08d2f8a87b/configure#L17913)
The script tries to use openssl by default, if it is build with openssl. If it is not build with openssl the script tries to use /dev/urandom. If nothing of the two are present you can use "--disable-strong-random" to use libpqs pseudo random generator (this is the way the configure script told me to do this).

In cross-compile environments the script will not detect /dev/urandom (https://github.com/postgres/postgres/blob/00b047fa67e9f4428a682bd90d5e6b08d2f8a87b/configure#L17870) so we have to use the pseudo-rng if not building with openssl.

So no, I dont think that this is a workaround. I think that this is the way to do it properly.

Reply:
Ok, now I understand and it makes sense. Thank you very much for this explanation!